doi-transformer/src/main/java/no/unit/nva/doi/transformer/CrossRefConverter.java

```diff
@@ -602,16 +588,15 @@ private URI extractFulltextLinkAsURI(CrossRefDocument document) {
      * @return An optional containing first link in list, without filtering.
      */
     private Optional<URI> extractFirstLinkToSourceDocument(CrossRefDocument document) {
-        // TODO Add filter to select what kind of document we want.
-        try {
-            List<Link> links = document.getLink();
-            if (nonNull(links) && !links.isEmpty()) {
-                return links.stream().findFirst().map(Link::getUrl).map(URI::create);
-            }
-        } catch (IllegalArgumentException e) {
-            logger.warn(MALFORMED_URL_MESSAGE);
-        }
-        return Optional.empty();
+        return hasLink(document.getLink()) ? getFirstLinkAsURI(document.getLink()) : Optional.empty();
```

Comment:
We call document.getLink() twice, can we avoid that? Possibly in the stream with a filter in getFirstLinkAsURI()

Reply:
getlink() moved up in the calling chain